src/task.c

```diff
@@ -357,9 +372,9 @@ static void ctx_switch(jl_task_t *lastt)
                 t->sticky = 1;
                 t->bufsz = 0;
                 if (always_copy_stacks)
-                    memcpy(&t->copy_stack_ctx, &ptls->copy_stack_ctx, sizeof(t->copy_stack_ctx));
+                    memcpy(&t->copy_stack_ctx, &ptls->copy_stack_ctx, sizeof(t->copy_stack_ctx) - sizeof(t->copy_stack_ctx.tsan_state));
```

Comment:
```suggestion
                    memcpy(&t->copy_stack_ctx.uc_mcontext, &ptls->copy_stack_ctx.uc_mcontext, sizeof(t->copy_stack_ctx.uc_mcontext));
```

But I'm not sure if `uc_mcontext` is something C can deal with, or if it needed to be wrapped in a struct to be something that C could create access.

Reply:
But note that this is in the `COPY_STACKS` code path, so currently we disable this code path entirely if TSAN is enabled